client/src/consumer.rs

```diff
@@ -402,8 +419,54 @@ impl<T: DeserializeMessage> MultiTopicConsumer<T> {
                 .map_err(|e| panic!("error creating referesh timer: {}", e))),
             subscription: subscription.into(),
             sub_type,
+            last_message_received: None,
+            messages_received: 0,
+            state_streams: vec![],
+        }
+    }
+
+    pub fn get_state_stream(&mut self) -> impl Stream<Item=ConsumerState, Error=()> {
```

Comment:
Not important at all, but  its always  weird for me to see a function like `get_XYZ` which takes `&mut self`.
I don't know if another verb could work like `register`, `create`, or `start`.

I can definitely live with it as is though if none of those sound better to you.